pkg/instance/repository.go

```diff
@@ -111,12 +112,17 @@ func (r repository) Delete(id uint) error {
 }
 
 func (r repository) FindByGroupNames(names []string, presets bool) ([]*model.Instance, error) {
-	var instances []*model.Instance
-
-	err := r.db.
+	query := r.db.
 		Preload("RequiredParameters.StackRequiredParameter").
-		Preload("OptionalParameters.StackOptionalParameter").
-		Where("group_name IN ?", names).
+		Preload("OptionalParameters.StackOptionalParameter")
+
+	isAdmin := slices.Contains(names, "administrators")
```

Comment:
Is this a group we create automatically and thus control the group name? If so it would be great to create a link between the code creating and the one here checking the group name. So both places use the exact same string 😄

Reply:
Yes, that is a group that we create automatically but that all takes place in the user service and is defined [here](https://github.com/dhis2-sre/im-user/blob/master/pkg/model/group.go#L9). It would actually be nice if it could be generated as part of the client but I'm not sure how and even if that make sense since the client itself is generated from our open api spec.

It is actually also defined [here](https://github.com/dhis2-sre/im-manager/blob/master/internal/handler/authorize.go#L8) but I don't want the repository to depend on something from the (internal) handler package.

Maybe we could just define it in the repository as it's defined in the handler package as noted above?